editoast/src/views/stdcm.rs

```diff
@@ -0,0 +1,216 @@
+use crate::core::stdcm::{
+    STDCMCoreRequest, STDCMCoreResponse, STDCMCoreRouteOccupancy, STDCMCoreStep,
+};
+use crate::core::{AsCoreRequest, CoreClient};
+use crate::error::Result;
+use editoast_derive::EditoastError;
+use serde::Serialize;
+use thiserror::Error;
+
+use crate::models::{
+    stdcm::{STDCMPath, STDCMRequestPayload, STDCMResponse},
+    Create,
+};
+
+use crate::models::{
+    Infra, Pathfinding, PathfindingChangeset, Retrieve, RollingStockModel, TrainSchedule,
+};
+
+use crate::DbPool;
+use actix_web::dev::HttpServiceFactory;
+use actix_web::web::{self, Data, Json};
+use actix_web::{post, HttpResponse, Responder};
+use diesel::PgConnection;
+
+use super::pathfinding::{
+    fetch_pathfinding_payload_track_map, parse_pathfinding_payload_waypoints, StepPayload,
+};
+use super::train_schedule::projection::Projection;
+use super::train_schedule::simulation_report::{create_simulation_report, SimulationReport};
+use super::train_schedule::{process_simulation_response, simulation_outputs_from_timetable_id};
+
+pub fn routes() -> impl HttpServiceFactory {
+    web::scope("/stdcm").service(create)
+}
+
+#[derive(Debug, Error, EditoastError, Serialize)]
+#[editoast_error(base_id = "stdcm")]
+#[allow(clippy::enum_variant_names)]
+enum StdcmError {
+    #[error("Infra {infra_id} does not exist")]
+    #[editoast_error(status = 400)]
+    InfraNotFound { infra_id: i64 },
+    #[error("Rolling stock with id {rolling_stock_id} doesn't exist")]
+    #[editoast_error(status = 400)]
+    RollingStockNotFound { rolling_stock_id: i64 },
+}
+
+#[post("")]
+async fn create(
+    db_pool: Data<DbPool>,
+    core: Data<CoreClient>,
+    data: Json<STDCMRequestPayload>,
+) -> Result<impl Responder> {
+    let stdcm_response = compute_stdcm(db_pool, core, data).await?;
+    Ok(HttpResponse::Created().json(stdcm_response))
+}
+
+async fn compute_stdcm(
+    db_pool: Data<DbPool>,
+    core: Data<CoreClient>,
+    data: Json<STDCMRequestPayload>,
+) -> Result<STDCMResponse> {
+    let core_output = call_core_stdcm(db_pool.clone(), &core, &data).await?;
+    let path = create_path_from_core_response(db_pool.clone(), &core_output, &data).await?;
+    let simulation =
+        create_simulation_from_core_response(db_pool, &core, &data, &core_output, &path).await?;
+    Ok(STDCMResponse { path, simulation })
+}
+
+async fn call_core_stdcm(
+    db_pool: Data<DbPool>,
+    core: &Data<CoreClient>,
+    data: &Json<STDCMRequestPayload>,
+) -> Result<STDCMCoreResponse> {
+    let conn = &mut db_pool.get()?;
+    let infra = Infra::retrieve(db_pool.clone(), data.infra_id)
```

Comment:
One of the other methods which is used at a couple of locations and could be factorized in a future pr.